### PR TITLE
License dataset

### PR DESCRIPTION
**Pull request by sergiplca**

### CODE REVIEW

File: src/main/java/cat/udl/eps/entsoftarch/repository/ClosedLicenseRepository.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+import java.util.List;
+
+/**
+ * Created by Victor on 03/03/2017.
+ */
+@RepositoryRestResource(collectionResourceRel="closedlicenses", path="closedlicenses")
+public interface ClosedLicenseRepository extends PagingAndSortingRepository <ClosedLicense, Long> {
+    List <ClosedLicense> findByText(@Param("text") String text, @Param("price") double price);
```
Comment (rogargon) [inline src/main/java/cat/udl/eps/entsoftarch/repository/ClosedLicenseRepository.java]: If the idea is to find by both text and price the method name should be findByTextAndPrice. More details: https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repository-query-keywords

If just find by text or price, the other method parameter should be omitted

File: src/main/java/cat/udl/eps/entsoftarch/repository/ClosedLicenseRepository.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.data.repository.query.Param;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+import java.util.List;
+
+/**
+ * Created by Victor on 03/03/2017.
+ */
+@RepositoryRestResource(collectionResourceRel="closedlicenses", path="closedlicenses")
```
Comment (rogargon) [inline src/main/java/cat/udl/eps/entsoftarch/repository/ClosedLicenseRepository.java]: Better to keep the default collectionResourceRel "closedLicense" to follow the same approach through the whole API

File: src/test/resources/features/RegisterLicense.feature [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    Given I login as "owner" with password "password"
+    And There are 0 open licenses registered
+    When I register a open license with text "text"
+    Then The new license has text "text"
+    And There is 1 open license registered
+
+  Scenario: Register a closed license with text and price
+    Given I login as "owner" with password "password"
+    And There are 0 closed licenses registered
+    When I register a closed license with text "text" and price 10
```
Comment (rogargon) [inline src/test/resources/features/RegisterLicense.feature:17]: What about ownership of the closed license? The idea is that it is associated to user "owner". Is it so?

**Review by rogargon — CHANGES_REQUESTED** [review]

File: src/test/resources/features/RegisterLicense.feature (hunk 3 above)
Comment (victorserrate) [reply src/test/resources/features/RegisterLicense.feature:17]: Hello Roberto, I don't understand what are you saying.  We replicated the same scenarios for both License types. What do you want exactly? Only the closed license will have owner or both?

Comment (rogargon) [reply src/test/resources/features/RegisterLicense.feature:17]: This is a scenario about closed license, but the same applies to open if they also have a field owner. Like in the register dataset feature, the idea is that one a user creates a license, the user becomes the owner. The scenario does not check this is true

**Review by rogargon — APPROVED** [review]

**Pull request merged**